src/ImageService.Data/ImageProjectRepository.cs

```diff
@@ -0,0 +1,55 @@
+﻿using LT.DigitalOffice.ImageService.Data.Interfaces;
+using LT.DigitalOffice.ImageService.Data.Provider;
+using LT.DigitalOffice.ImageService.Models.Db;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace LT.DigitalOffice.ImageService.Data
+{
+    public class ImageProjectRepository : IImageProjectRepository
+    {
+        private readonly IDataProvider _provider;
+
+        public ImageProjectRepository(IDataProvider provider)
+        {
+            _provider = provider;
+        }
+
+        public List<Guid> Create(List<DbImagesProject> imagesProjects)
+        {
+            if (imagesProjects.Contains(null))
+            {
+                throw new ArgumentNullException(nameof(imagesProjects));
+            }
+
+            _provider.ImagesProjects.AddRange(imagesProjects);
+            _provider.Save();
+
+            return imagesProjects.Select(x => x.Id).ToList();
+        }
+
+        public bool Delete(List<Guid> imageIds)
+        {
+            if (imageIds == null)
+            {
+                return false;
+            }
+
+            _provider.ImagesProjects.RemoveRange(Get(imageIds));
+            _provider.Save();
+
+            return true;
+        }
+
+        public List<DbImagesProject> Get(List<Guid> imageIds)
+        {
+            return _provider.ImagesProjects.Where(x => imageIds.Contains(x.Id)).ToList();
+        }
+
+        public DbImagesProject Get(Guid imageId)
+        {
+            return _provider.ImagesProjects.Find(imageId);
```

Comment:
use first or default. Do you know how work find? What result will be if we don't have required record?

Reply:
done